pkg/k8sclient/k8sclient.go

```diff
@@ -251,12 +251,17 @@ func parsePodNetworkAnnotation(podNetworks, defaultNamespace string) ([]*types.N
 	return networks, nil
 }
 
-func getKubernetesDelegate(client *ClientInfo, net *types.NetworkSelectionElement, confdir string, pod *v1.Pod, resourceMap map[string]*types.ResourceInfo) (*types.DelegateNetConf, map[string]*types.ResourceInfo, error) {
+func getKubernetesDelegate(client *ClientInfo, net *types.NetworkSelectionElement, confdir string, pod *v1.Pod, resourceMap map[string]*types.ResourceInfo, try bool) (*types.DelegateNetConf, map[string]*types.ResourceInfo, error) {
 
-	logging.Debugf("getKubernetesDelegate: %v, %v, %s, %v, %v", client, net, confdir, pod, resourceMap)
+	logging.Debugf("getKubernetesDelegate: %v, %v, %s, %v, %v, %v", client, net, confdir, pod, resourceMap, try)
 	customResource, err := client.NetClient.NetworkAttachmentDefinitions(net.Namespace).Get(context.TODO(), net.Name, metav1.GetOptions{})
 	if err != nil {
 		errMsg := fmt.Sprintf("cannot find a network-attachment-definition (%s) in namespace (%s): %v", net.Name, net.Namespace, err)
+		if try {
+			// just an attempt to find NAD in getNetDelegate
+			// do not generate event
+			return nil, resourceMap, fmt.Errorf("getKubernetesDelegate: " + errMsg)
+		}
```

Comment:
I suppose we should not introduce new arugment `try` in this case because this argument may change the behavior of the function.